src/main/java/seedu/recipe/ui/RecipeCard.java

```diff
@@ -50,45 +60,79 @@ public class RecipeCard extends UiPart<Region> {
 
     /**
      * Creates a {@code RecipeCode} with the given {@code Recipe} and index to display.
+     * 
+     * @param recipe the {@code Recipe} to display
+     * @param displayedIndex the index of the {@code Recipe} in the list
      */
     public RecipeCard(Recipe recipe, int displayedIndex) {
         super(FXML);
         this.recipe = recipe;
+        cardPane.setFocusTraversable(true);
         id.setText(displayedIndex + ". ");
         name.setText(recipe.getName().recipeName);
 
         //Duration
-        duration.setText(
+        duration.setText("Duration: " +
                 Optional.ofNullable(recipe.getDurationNullable())
                         .map(Object::toString)
                         .orElse("Duration was not added.")
         );
 
         //Portion
-        portion.setText(
+        portion.setText("Portion: " +
                 Optional.ofNullable(recipe.getPortionNullable())
                         .map(Object::toString)
                         .orElse("Portion was not added.")
         );
 
-        //Tags
-        recipe.getTags().stream()
-                .sorted(Comparator.comparing(tag -> tag.tagName))
-                .forEach(tag -> tags.getChildren().add(new Label(tag.tagName)));
-
         //Ingredients
+        ingredientsTitle.setText("Ingredients:");
         recipe.getIngredients()
                 .forEach(ingredient -> ingredients.getChildren().add(new Label(ingredient.toString())));
-
+        
         //Steps
+        stepsTitle.setText("Steps:");
         recipe.getSteps()
                 .forEach(step -> steps.getChildren().add(new Label(step.toString())));
 
+        //Tags
+        recipe.getTags().stream()
+            .sorted(Comparator.comparing(tag -> tag.tagName))
+            .forEach(tag -> tags.getChildren().add(new Label(tag.tagName)));        
+
+        //Selector focus
+        cardPane.setOnMouseEntered(event -> {
+            cardPane.requestFocus();
+        });
+
         // Add a click listener to the cardPane node
         cardPane.setOnMouseClicked(event -> {
+            cardPane.requestFocus();
             RecipePopup popup = new RecipePopup(recipe, displayedIndex);
             popup.display();
         });
+
+        // Handle keypress events
+        cardPane.setOnKeyPressed(event -> {
+            cardPane.requestFocus();
+            KeyCode input = event.getCode();
+            if (input == KeyCode.DELETE ||
+                input == KeyCode.D ||
+                input == KeyCode.BACK_SPACE) {
+                DeleteRecipeEvent deleteEvent = new DeleteRecipeEvent(displayedIndex);
```

Comment:
Good consideration on different keys such as backspace and delete for deleting a recipe- makes life easier especially for good typists that don't have to navigate with their mouse